weblate/trans/models/translation.py

```diff
@@ -439,6 +440,18 @@ def check_sync(self, force=False, request=None, change=None) -> None:  # noqa: C
                                 unit.source = translated_unit.source
                         except UnitNotFoundError:
                             pass
+                    if (
+                        self.component.file_format_cls.monolingual
+                        and self.component.key_filter_re
+                        and re.match(self.component.key_filter_re, unit.context) is None
```

Comment:
The expression is already compiled, no need to call re.match on it:

```suggestion
                        and self.component.key_filter_re.match(unit.context) is None
```